### ISSUE TITLE
RFC Comments v1.0.0

### ISSUE THREAD

**Issue by egabancho** [inveniosoftware/invenio-comments#13]
WebComment is the legacy commenting engine which allow users to add comments to records in the form of a discussion.

The current main features are:
- [ ] Replay to comment
- [ ] WYSIWYG/Markdown editor
- [ ] Filtering by text
- [ ] Filtering by file and version association
- [ ] Attach files to comments
- [ ] Associate comments to record files and versions
- [ ] Subscription to a comment
- [ ] Soft commenting deadline
- [ ] Custom extra headline for comments
- [ ] Collaps expand comments (persistent per user)
- [ ] Report abuse (+ admin tools)
- [ ] Subscribe to the whole discussion
- [ ] Subscribe to a single comment (should we subscribe users to their own comments by default ?)
- [ ] Post replies to comments over email

It would be nice if the new implementation is more flexible allowing comments to be connected to a document (or anything) directly. 

One possible solution would be to build an extension that uses any of the existing commenting engines like [isso](https://github.com/posativ/isso/)(written in python) or [Disqus](https://disqus.com).

But I think we should also allow custom implementations as some of the functionalities we are using right now are quite hard to find.

cc @kasioumis 

Note: moved from https://github.com/inveniosoftware/invenio/issues/2957

**Comment by jirikuncar** [comment]
> Subscribe to a single comment

Do you mean comment thread?

**Comment by egabancho** [comment]
Yes, for example I want to receive a notification if somebody posts a reply to a comment I made but not from a general comment in the main thread
